kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/ButtonAlRichMessage.java

```diff
@@ -0,0 +1,72 @@
+package com.applozic.mobicomkit.uiwidgets.conversation.richmessaging;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.applozic.mobicomkit.api.conversation.Message;
+import com.applozic.mobicomkit.uiwidgets.AlCustomizationSettings;
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.callbacks.ALRichMessageListener;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.models.ALRichMessageModel;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.views.KmFlowLayout;
+import com.applozic.mobicommons.json.GsonUtils;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ButtonAlRichMessage extends AlRichMessage {
+
+    public ButtonAlRichMessage(Context context, LinearLayout containerView, Message message, ALRichMessageListener listener, AlCustomizationSettings alCustomizationSettings) {
+        super(context, containerView, message, listener, alCustomizationSettings);
+    }
+
+    @Override
+    public void createRichMessage() {
+        super.createRichMessage();
+        KmFlowLayout flowLayout = containerView.findViewById(R.id.kmFlowLayout);
+        setupAlRichMessage(flowLayout, model);
```

Comment:
as flowLayout is not getting used anywhere so we can directly write:
        setupAlRichMessage(containerView.findViewById(R.id.kmFlowLayout), model);

Reply:
done